### ISSUE TITLE
Tailoring Komi for KataGo intentional aggressive overplay with weaker/kyu humans?

### ISSUE THREAD

**Issue by bxc2739** [lightvector/KataGo#644]
Hi this is not really an issue or a bug but more of a general quick question...

Is there a link or documentation or a quick setting or  parameter I can tune with KataGo AI to have it play super aggressively when I know its playing against weak players (such as myself) at say 15kyu level to 25 kyu level? A experienced Dan player would be able to "clear the board" if he choose to play aggressive with a beginner and yet AI bots who are stronger do not do this...

Is setting Komi the only way to emulate this? and if so what is the best way to go about this?

**Comment by lightvector** [comment]
The first thing would be to make sure playoutDoublingAdvantage is set to 3.0, the most extreme possible value of that parameter. This makes KataGo behave the way it does in training when playing with a copy of itself that searches less deep - hence likely to favor more tactical complications.

The next thing to do would probably be to increase dynamicScoreUtilityFactor mildly, from 0.3 to maybe 0.5 or 0.6, to increase the value assigned to catching up on points.

After that, dynamically adjusting komi to keep the winrate around 5-20% could help, although you'd need to test it.

However, KataGo is probably not going to "clear the board" against beginners that play very defensively. KataGo plays fairly honestly and properly. It tries its best to play the "objective" best move that gets a good result regardless if the opponent responds perfectly. So if it thinks move A is objectively better than move B, then it will play move A, even if move B would make it more likely for a double-digit-kyu player to make a huge blunder.

In other words, KataGo tries not to make intentionally trick plays or overplays or other bad moves that are good against weak players but in realty are bad moves that you shouldn't learn or emulate because they will stop working as you start to face opponents as strong as you are. A big reason an experienced dan player might be able to "clear the board" is due to precisely this - playing trick moves and overplays, sometimes even *unintentionally* due to not knowing that an overplay really is too far - but where a player at a top pro or even higher level would know better.
